pkg/idp/idp.go

```diff
@@ -96,34 +96,31 @@ func (s *Service) Reconcile(ctx context.Context) error {
 	}
 	{
 		// Get existing connectors from the dex config secret
-		oldConnectors, err := getConnectorsFromSecret(secret)
+		oldConfig, err := getDexConfigFromSecret(secret)
 		if err != nil {
 			return microerror.Mask(err)
 		}
+		oldConnectors := getConnectorsFromConfig(oldConfig)
 		// Create apps for each provider and get dex config
 		appConfig, err := s.GetAppConfig(ctx)
 		if err != nil {
 			return microerror.Mask(err)
 		}
-		dexConfig, err := s.CreateOrUpdateProviderApps(appConfig, ctx, oldConnectors)
+		newConfig, err := s.CreateOrUpdateProviderApps(appConfig, ctx, oldConnectors)
```

Comment:
I think we can use this config as the new data source directly. It does not need to be serialized and deserialized again.